ProvisionSample/Program.cs

```diff
@@ -588,29 +594,65 @@ static void Main(string[] args)
             return client;
         }
 
+        static async Task<IEnumerable<string>> GetTenantIdsAsync(string commonToken)
+        {
+            using (var httpClient = new HttpClient())
+            {
+                httpClient.DefaultRequestHeaders.Add("Authorization", "Bearer " + commonToken);
+                var response = await httpClient.GetStringAsync("https://management.azure.com/tenants?api-version=2016-01-29");
```

Comment:
I see that GetAzureAccessTokenAsync checks for this returning a null/empty response, and in that case throws an exception that it couldn't get the tenant. 

Let's consider the exceptions that could occur in GetTenantIdsAsync (network failure, non-success response codes, success responses that don't have the expected "value"/"tenantId" tokens) - do we want to catch those and put them down the same flow of "couldn't get the tenant," or have them throw up the stack as is?